cmd/main.go

```diff
@@ -207,8 +209,11 @@ func main() {
 		}
 
 		mgr = mcmgr.GetLocalManager()
+		metrics.Registry.MustRegister(&dnsMetrics.RemoteClusterCollector{Provider: clusterProvider})
 	}
 
+	metrics.Registry.MustRegister(&dnsMetrics.LocalCollector{Ctx: ctx, Mgr: mgr})
```

Comment:
Do I get this correctly:
All we need to do is register a collector, and it will `Collect()` on each scrape? 
Meaning that we don't need to do anything else inside the code. How often will the `Collect()` happen (my concern is that it is a lot of `.List()` calls)?